lib/application.js

```diff
@@ -534,16 +532,8 @@ app.render = function render(name, options, callback) {
     opts = {};
   }
 
-  // merge app.locals
-  merge(renderOptions, this.locals);
-
-  // merge options._locals
-  if (opts._locals) {
-    merge(renderOptions, opts._locals);
-  }
-
   // merge options
-  merge(renderOptions, opts);
+  var renderOptions = { ...this.locals, ...(opts._locals || {}), ...opts };
```

Comment:
AFAIK there is no need to have a fallback, spreading a variable that is `undefined` results in an empty object:

```js
var a = undefined
var b = { ...a }
a // {}
```

```suggestion
  var renderOptions = { ...this.locals, ...opts._locals, ...opts };
```

Reply:
Thanks